app/src/main/java/lt/getpet/getpet/authentication/AuthenticationManager.kt

```diff
@@ -108,6 +109,12 @@ class AuthenticationManager(
         }
     }
 
+    fun signOutUser(context: Context): Single<Unit> {
+        return Single.just(AuthUI.getInstance().signOut(context)).map {
+            appPreferences.apiToken.delete()
+        }
+    }
```

Comment:
Possibly there is no need to execute these operations on different thread, so I would avoid RxJava.

Reply:
After removing threading and testing it on my device it seems that it is necessary to use threading here. 

What in that case happened, if I logged out and then tried to sign in again, for a split second my old profile would show. Maybe the operation was not fully done when I signed out which caused this flash. 

So I decided to leave threading here in the end.